src/features/upload-file/factories/delete-file-factory.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { ActionRequest, RecordActionResponse, ActionContext, BaseRecord } from 'admin-bro'
```

Comment:
to slim down main feature function I extracted all other functions creating hooks, etc to factoryFunction. Each of them has the same params: (UploadOptions and provider). So now I can easily test them.